docs/DeveloperGuide.md

```diff
@@ -1,34 +1,124 @@
+
+- Alternative 1 (current choice): Object-oriented style with a separate class for <code>DeleteCommand</code>
+ 
+  - Pros: Easy to add the delete feature without having to change the logic of the code much as each command object
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement delete feature in the <code>Duke</code> class
+
+  - Pros: Will have lesser code to deal with as a new function is simply created in the <code>Duke</code> class
+  
+  - Cons: Code becomes less organised since for every other command that we have implemented, <code>Duke</code> class
+  simply executes those commands as black boxes, without worrying about their internal details
+
+### 2.2 Set budget feature
+#### 2.2.1 Current implementation
 
-## Product Scope
+The set budget feature is implemented using a <code>SetBudgetCommand</code> class which extends the main
+<code>Command</code> class with a variable representing the budget amount.
+
+The <code>Duke</code> class first receives user input from the <code>Ui</code> class before it creates a 
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
+<code>SetBudgetCommand</code> object based on that user input.
+
+The <code>Duke</code> class then calls the <code>execute</code> method of the <code>SetBudgetCommand</code> object
+which makes another call to the <code>setBudget</code> function of the <code>Budget</code> object 
+with the amount specified by the user for the budget.
+
+The following sequence diagram below shows how the set budget feature works. Note the <code>Ui</code> class is
+omitted in the sequence diagram to emphasise on the other classes:
+
+![alt text](images/SetBudget.png)
+
+
+#### 2.2.2 Design considerations
+
+##### Aspect: Data structure to support the set budget feature
+
+- Alternative 1 (current choice): Object-oriented style with a separate class for <code>SetBudgetCommand</code>
+ 
+  - Pros: Easy to add the set budget feature without having to change the logic of the code much as each command object
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement set budget feature in the <code>Duke</code> class
+
+  - Pros: Will have lesser code to deal with as a new function is simply created in the <code>Duke</code> class
+  
+  - Cons: Code becomes less organised since for every other command that we have implemented, <code>Duke</code> class
+  simply executes those commands as black boxes, without worrying about their internal details
+
+
+## Appendix A: Product Scope
 ### Target user profile
 
-{Describe the target user profile}
+- Likes to cook at home and requires help keeping track of complex grocery shopping lists and
+staying within budget
+- Prefers to use computer applications as opposed to mobile applications
```

Comment:
Maybe can change to something like, "Prefers to use Command Line Interface apps as opposed to other kinds of apps or physical paper". Cause we might want to let users use it on mobile if possible since its more convenient

Reply:
Ok will do!